### DISCUSSION TITLE
How can you test storage (bucket) policies?

### DISCUSSION THREAD

**Discussion by tcaish** [supabase/supabase#20286] [Questions] [answered]
I'm setting up my storage buckets, and I'm adding policies to each bucket, but I have no way to test any operations on the bucket. There is no way to change the role like we can for database queries.

For example, if I wanted to test my bucket policy that only authenticated users can upload a file to the bucket, how can I do that from the website?

**Comment by GaryAustin1** [comment, marked as answer]
You can mint a jwt with needed info including role and sub if authenticated.  Then put the jwt in the authorization header and make calls as that user or role.  Really only two roles that matter for RLS and that is anon and authenticated.  https://supabase.com/docs/learn/auth-deep-dive/auth-deep-dive-jwts#jwts-in-supabase

**Reply by tcaish** [reply]
This is a good workaround, so thank you for that! As a future feature of Supabase, it would be nice to be able to test these policies with a given role like you can for the database via the website.

**Comment by rafal-r** [comment]
You can also test them like any other RLS policies using pgTap https://supabase.com/docs/guides/database/testing
